File: angle-chan-server/src/main/java/com/ben/dektor/entities/MediaContainer.java
package com.ben.dektor.entities;

import lombok.Data;

import javax.persistence.*;

@Entity
@Data
public class MediaContainer {
    @Id @GeneratedValue
private Long id;
    @OneToOne(cascade = CascadeType.ALL)
private MediaInfo info;
    @OneToOne(cascade = CascadeType.ALL)
private Thumbnail thumbnail;
private String mediaSizeKb;
private String title;
private String mediaType;
}

File: angle-chan-server/src/main/kotlin/com/ben/dektor/entities/MediaContainer.kt
package com.ben.dektor.entities

import lombok.Data
import javax.persistence.*

@Entity
data class MediaContainer (
    @Id
    @GeneratedValue
    var id: Long?,

    @OneToOne(cascade = [CascadeType.ALL])
    var info: MediaInfo,

    @OneToOne(cascade = [CascadeType.ALL])
    var thumbnail: Thumbnail,
    var mediaSizeKb: String?,
    var title: String?,
    var mediaType: String?
    )